### DISCUSSION TITLE
Is anyone else experiencing generated SQL issues?

### DISCUSSION THREAD

**Discussion by MackenzieStorey** [hasura/graphql-engine#9421] [General] [answered]
This is moreso for my sanity to double check I did not change a setting improperly. When generating a query using the `_or` clause, my generated SQL is using an `AND`. I am attempting to retrieve data that requires an OR condition, but my automated SQL is spitting back an AND. Here are some pics.

GraphQL query
![image](https://user-images.githubusercontent.com/77168117/217334834-f831feb2-6bed-4855-948e-7a8ecec154dc.png)

Generated SQL
![image](https://user-images.githubusercontent.com/77168117/217335181-7efc8cd0-0b3c-4e96-b624-d9dc318f1657.png)

Anyone else seeing this?

**Comment by meetzaveri** [comment, marked as answer]
Hey @MackenzieStorey 

From what I am seeing from your graphql query, it seems that `_or` condition is never in match with that of first clause (before comma seperated)

There are two conditions here in "seperate" manner:
1. Equality check, where it checks if `userId` equals to given value.
2. A separate clause of `_or` where it contains the same condition as (1)

So because of these two clauses being seperate conditions, it results into `AND` in generated SQL .

### Rectifiction to your use-case query:
I get it that from syntax of introspection query, it looks confusing. But if your use-case was to get results where `userId` equals to either one or another value, you can solely use `_in` operator i.e. list based search operator against `userId`. And in variable, pass values into array. 

Docs on list based operators usage here - https://hasura.io/docs/latest/queries/postgres/query-filters/#list-based-search-operators-_in-_nin

Example query
```gql
query getArticlesBySpecificAuthorIds( $list_of_ids: [uuid!] = "") {
  articles(where: {author_id: {_in: $list_of_ids}}) {
    title
    author_id
  }
}
```

variables
```json
{
"list_of_ids":
  [
  "56364b29-aa14-4ec9-b129-4d87d5b5da80",
  "71229f10-55a1-4861-9c81-56f4b1d68a58"
  ]
}
```
### Alternate solution if you want to stick with `_or`
If you wish to use `_or` operator for comparing one or more value, then alternate example can also look like

```gql
query getArticlesBySpecificAuthorIds {
  articles(where: {
      _or: [
        {author_id: {_eq: "56364b29-aa14-4ec9-b129-4d87d5b5da80"}},
        {author_id: {_eq: "71229f10-55a1-4861-9c81-56f4b1d68a58"}}
      ]
    }
  ) {
    title
    author_id
  }
}
```

where generated SQL would look like
```sql
FROM
          "public"."articles"
        WHERE
          (
            (
              ("public"."articles"."author_id") = (('56364b29-aa14-4ec9-b129-4d87d5b5da80') :: uuid)
            )
            OR (
              ("public"."articles"."author_id") = (('71229f10-55a1-4861-9c81-56f4b1d68a58') :: uuid)
            )
          )
 ```

I hope this helps clarifying the situation here. Do let me know otherwise.
